src/lib/ui/ClimateSettingForm/ClimateSettingScheduleDeviceSelect.tsx

```diff
@@ -0,0 +1,29 @@
+import { isThermostatDevice } from 'seamapi'
+
+import { DeviceTable, type UseDevicesData } from 'lib/index.js'
+
+interface ClimateSettingScheduleDeviceSelectProps {
+  onSelect: (deviceId: string) => void
+}
+
+export function ClimateSettingScheduleDeviceSelect({
+  onSelect,
+}: ClimateSettingScheduleDeviceSelectProps): JSX.Element {
+  const deviceFilter = (
+    device: UseDevicesData[number],
+    searchInputValue: string
+  ): boolean => {
+    if (!isThermostatDevice(device)) return false
+    const value = searchInputValue.trim()
+    if (value === '') return true
+    return device.properties.name.toLowerCase().includes(value)
```

Comment:
Can we re-use the default filter? Maybe we should export it.
```suggestion
    return defaultDeviceFilter(...)
```

Reply:
<img width="1098" alt="Screenshot 2023-09-06 at 5 16 58 PM" src="https://github.com/seamapi/react/assets/11382084/a699bc36-ab16-4b1f-a32c-c61a1453d582">

Should this even be something you can override? You need this for search to work. 

Perhaps the device table should allow additional filtering that doesn't override this part

Reply:
I think we have to let them override, but if we export this function then they can re-use it in their filter.

Reply:
[7771f42](https://github.com/seamapi/react/pull/440/commits/7771f42a0df36d12d6d2d51af2a8b72a9e70e61f)